lib/storage.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const data = {};
+const Promise = require('bluebird');
+const storage = module.exports = {};
+const dataDir = `${__dirname}/../data`;
+const fs = Promise.promisifyAll(require('fs'));
+
+storage.setItem = function(name, item) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    err.status = 400;
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(item);
+    return fs.writeFileAsync(`${dataDir}/${name}/${item.id}.json`, json);
+  })
+  .then(() => {
+    return item;
+  });
+};
+
+storage.getItem = function(name, id) {
+  return fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(id);
+    return fs.readFileAsync(`${dataDir}/${name}/${id}.json`, json);
+  })
```

Comment:
The fs.readFileAsync isn't quite right and does not properly return the item. Research the docs again, but remember, we've wrapped the fs. methods in a Promise, so the correct syntax will not exactly match the docs. 

I can help you out with this whenever you have time if you still have trouble. :)